subsys/bluetooth/mesh/rpl.c

```diff
@@ -344,3 +348,39 @@ void bt_mesh_rpl_pending_store(uint16_t addr)
 		}
 	}
 }
+
+size_t bt_mesh_rpl_get(void (*cb)(const struct bt_mesh_rpl *rpl_entry))
+{
+	struct bt_mesh_rpl entry;
+	size_t count = 0;
+
+	if (!cb) {
+		return 0;
+	}
+
+	for (int i = 0; i < ARRAY_SIZE(replay_list); i++) {
+		if (replay_list[i].src) {
+			entry = replay_list[i];
+		} else {
+			break;
+		}
+
+		cb(&entry);
```

Comment:
I do not see how we can calculate size of the RPL list to make sure we have enough space (and time) needed for storing the RPL list. This has to be put in the application somehow. The goal was to keep the emergency data storage independent of any other modules.
I also wonder will this store every entry as separate entity, and not the hole RPL list as one entity.

Reply:
To calculate size and time it requires iterating through the list using this function and callback. When the function is left the returned value provides the total number of busy cells in the RPL. The application can create its own data array for storing from data provided via the callback and calculate the necessary time and space. This is not necessarily required to store RPL as is. For example, currently, only 2 parameters go to the settings. 
```
struct rpl_val {
	uint32_t seq:24,
	      old_iv:1;
};
```
Yep, it requires some application code that uses this API. To store RPL as a bunch of separate cells or as a single entity is up to the application.

Reply:
If does not really only store 2 parameters, it stores the seq and old_iv as data, but the src is stored in the filename for settings. So in settings the seq and old_iv is stored for every src.